### DISCUSSION TITLE
How is the repeat mode implemented?

### DISCUSSION THREAD

**Discussion by lhughey** [Tapanila/SharpCaster#305] [Q&A] [answered]
I'm looking at repeating the playback of a single file, but can't figure out how to implement it. I took a look at the documentation at https://developers.google.com/cast/docs/reference/web_sender/chrome.cast.media#.RepeatMode and see that it is a property of the media, so I tried passing it as a customdata property with no luck, like this

            `var media = new Media
            {
                ContentUrl = fileToPlay
            };
            media.CustomData = new Dictionary<string, string>();
            media.CustomData.Add("RepeatMode", "ALL");

            MediaStatus = await CastClient.GetChannel<IMediaChannel>().LoadAsync(media);`

I also tried to pass it by setting the media status, with no luck, like this 
`MediaStatus.RepeatMode = "ALL";`
 
I also picked up the string "REPEAT_ALL" from somewhere (not sure).

Can you tell me what I'm doing wrong? Thanks.

**Comment by Tapanila** [comment, marked as answer]
Seems like I hadn't implemented it.

Looking at the documentation it looks like setting repeat mode on load is only implemented at queue load message:
https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.messages.QueueLoadRequestData

Because this call is used for loading media:
https://developers.google.com/cast/docs/reference/web_receiver/cast.framework.messages.LoadRequestData

I added the implementation for this in 1.1.18 and created a test case for it:

            var media = new Media
            {
                ContentUrl = "https://incompetech.com/music/royalty-free/mp3-royaltyfree/Loping%20Sting.mp3"
            };

            var queueItem = new QueueItem
            {
                Media = media
            };


            await client.GetChannel<IMediaChannel>().QueueLoadAsync([queueItem], null, RepeatModeType.ALL);
            var test = await client.GetChannel<IMediaChannel>().PlayAsync();

**Comment by lhughey** [comment]
Thanks for the quick response and update! This worked.
